gme/Spc_Emu.cpp

```diff
@@ -14,7 +14,7 @@
 #define UINT unsigned int
 #define LONG long
 #define HANDLE void *
-#define LPARAM intptr_t
+#define LPARAM uintptr_t
```

Comment:
This is wrong: LPARAM is typedef'ed as LONG_PTR in win32 sdk, therefore intptr_t has been the correct solution here.

Reply:
`LPARAM` can be whatever we need it to be to suit our purposes. The default value for `LPARAM` in dll.hpp is `long`. That didn't suit my purposes for the callback, so I changed it to `intptr_t`, but that's actually not the most conventional type for storing pointer values. `uintptr_t` is. Unless we need to represent negative pointer values, we should avoid `intptr_t`.

Reply:
What I can do is change the placement of `LPARAM` so that it's non OS-specific.

Reply:
> The default value for `LPARAM` in dll.hpp is `long`

Because it (possibly rightfully) assumes that all unices are either ILP32 or LP64

What are you really gaining by switching to uintptr_t?

Reply:
Do our pointer values need to be signed? Do we need to be able to represent negative pointer values? If not, I would say that switching to `uintptr_t` does a better job of spelling out the intended use of those parameters.

Reply:
The thing is, the intention is at unrar library side and not ours: we aren't in control of it. The library already builds itself with LPARAM as LONG_PTR or long, so the question still stands as to what are you really gaining from this change.

Reply:
Probably nothing from a functional standpoint. The change was supposed to help clarify what the intended use was on our end. Pointer values are represented as unsigned integers.

My understanding is they can be greater than the maximum value of signed long. In theory, maybe it could result in an especially large pointer value being misinterpreted? I'm not sure.

Reply:
Wait a second, I seem to recall that the whole reason `long` didn't work is because windows actually uses `long long`. Have I got that right? I seem to remember running into that issue when building on Windows with MSYS2.

The history of LPARAM being signed seems to be that it was a signed 32-bit integer when Windows was 16-bit. Then when Windows became 32-bit, it became a signed 64-bit. They've been maintaining backward compatibility by making the integer used to represent the pointer value bigger than the actual pointer. In which case, we definitely should use `uintptr_t`, and using `intptr_t` could have actually resulted in undefined behavior.

Reply:
unrar is using LPARAM as long _only_ for unix. For windows, it uses LPARAM as LPARAM i.e. LONG_PTR as defined in win32 sdk. libgme before commit ed4ed86f5bb71d313176e798d3f8559a1f1d2af9 was broken on windows (libgme's fault, not unrar's)

Reply:
Yes but technically it should be `unsigned long`, shouldn't it?
It would have to be an absolutely enormous memory address value to make a difference, implying that there are several exabytes of ram on the computer.

Also, I looked into it, Windows doesn't use `long long`, I was wrong about that. It's probably for that reason, that it would take unbelievably large amounts of ram for that to become an issue.